common/src/main/java/com/wynntils/services/mapdata/attributes/type/MapAttributes.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.mapdata.attributes.type;
+
+import com.wynntils.utils.colors.CustomColor;
+import com.wynntils.utils.render.type.TextShadow;
+
+public interface MapAttributes {
+    // If this is the empty string (""), then no label will be displayed
+    // null means inherit
+    String getLabel();
+
+    // If this is MapFeatureIcon.NO_ICON_ID ("none"), then no icon will be displayed
+    // null means inherit
+    String getIconId();
+
+    // 1-1000, 1000 is highest priority (drawn on top of everything else)
+    // 0 means no value specified; inherit
+    int getPriority();
+
+    // 0 means no information is available, or suitable for all combat levels
```

Comment:
Shouldn't this be split into 2 different values then..? Kind of weird, but imagine filtering for all POIs "my level", perhaps I don't want to include unspecified level ranges... Not too sure on how/if to fix the problem though

Reply:
That is so heavily dependent on the data source. Let's say someone publishes a list of good places to grind mobs for an unusual ingredient. They might not bother to specify level ranges, but that does not mean it is suitable for all levels. Or if someone publishes a list of fancy names they are giving all entrances to Ragni. (Why not..?) Then it would not even be really clear if that would be suitable for all levels, or if the concept of level does not really apply.

I agree that it is a bit odd if you want to filter on a certain level. But it's kind of like when you sort quests based on distance -- what to do with the quests that do not have a location? In legacy, we used to sort them first as if they were the nearest quest. This was probably wrong, but so would sort them last be.

Reply:
I only worry about this since this is a design decision we have to "fix" now, if we ever want to. Unless you want to go down the versioning rabbit hole..? Btw, does this support any kind of versioning when parsing? That might be simple but really useful to explore.

Reply:
That's a valid point. We might want to prescribe a version info for the json files...

Reply:
I agree that this decision needs to be fixed now. I just don't understand how we could separate map features that do not have level information from "do not apply" to "we do not know". Let's take the Item Identifier in Llevigar for instance. He has no level information. Should he inherit the level information of Llevigar, since in practice, you should not go there unless you're of sufficient level? Or should he be marked "accessible for all", since every level can in fact interact with him? But then again, in *theory* even a level 1 character can go to Llevigar, it's just a bloody stupid idea.

Reply:
This is 2 separate issues though.

1, The level "code" itself has to support 3 different values: actual integer, "all levels", "unknown".
2, We have to figure out what to put here when implementing the default providers. That is where the inheritance would come in (which I think is appropriate in your example).

This PR, we only basically need to care about the first issue, if we should support it, and if how.

Reply:
Or take Light's Secret. It is missing level indicator on the official map. But that is really mostly an outage, it should be marked as level ~70. (Just as Ozoth's Spire). In this case, the absense of level information does not mean "accessible for all", but "we do not know", or even "should have had level information, but it is missing in the source material".

The trouble is, there is no way we can systematically and programmatically differentiate between these various reasons for a map feature missing level information. So the separation is between "has indication of level" and "do not have indication of level" -- and the reason for the former can never be known to us.

The latter state is preferably represented as 0, rather than having an Optional<Integer>, for reasons of simplicity. Since characters start out at level 1, level 0 is perfectly usable as an indicator for "does not have level information". With that said, the field is optional in json files, in which the value 0 is assumed.

Reply:
Perhaps we could just make -1 "accessible for all"?

Reply:
Let's assume we do that. How should we present this on the map?